samples/tab-channel-group-config-page-auth/csharp/README.md

```diff
@@ -0,0 +1,92 @@
+
+## Pre-requisites
+
+- Microsoft Teams is installed and you have an account (not a guest account)
+- [.NET Core SDK](https://dotnet.microsoft.com/download) version 3.1
+
+  ```bash
+  # determine dotnet version
+  dotnet --version
+  ```
+- [ngrok](https://ngrok.com/) or equivalent tunnelling solution
+
+- Visual Studio
+
+- Asp.net Core
+
+## To try this sample
+
+> Note these instructions are for running the sample on your local machine, the tunnelling solution is required because
+
+1) Clone the repository
+    git clone https://github.com/Microsoft/TabAuthentication.git
+
+2) If you are using Visual Studio
+    - Launch Visual Studio
+    - File -> Open -> Project/Solution
+    - Navigate to `TabAuthentication` folder
+    - Select `TabAuthentication.csproj` file
+    - Press `F5` to run the project
+
+3) Run ngrok - point to port 3978
+    ```bash
+    ngrok http -host-header=rewrite 3978
+    ```
+4) Update the `appsettings.json` configuration for the tab to use the Microsoft App Id in TabAuthentication folder, App Password and Connection Name from the Bot Framework           registration. (Note the App Password is referred to as the "client secret" in the azure portal and you can always create a new client secret anytime.)
+
+### Register an Azure AD Application
+
+  Your tab needs to run as a registered Azure AD application in order to obtain an access token from Azure AD. In this step you'll register the app in your tenant and give Teams   permission to obtain access tokens on its behalf.
+
+  Create an [AAD application](https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/auth-aad-sso#1-create-your-aad-application-in-azure) in           Azure. You can do this by visiting the "Azure AD app registration" portal in Azure.
+
+1) Set your application URI to the same URI you've created in Ngrok.
+   - Ex: api://contoso.ngrok.io/{appId} using the application ID that was assigned to your app
+          
+2) Setup your redirect URIs. This will allow Azure AD to return authentication results to the correct URI.
+   - Visit Manage > Authentication.
+   - Create a redirect URI in the format of: https://contoso.ngrok.io/auth/auth-end.
+   - Enable Implicit Grant by selecting Access Tokens and ID Tokens.
+          
+3) Setup a client secret. You will need this when you exchange the token for more API permissions from your backend.
+   - Visit Manage > Certificates & secrets
+   - Create a new client secret.
+          
+4) Setup your API permissions. This is what your application is allowed to request permission to access.
+   - Visit Manage > API Permissions
+   - Make sure you have the following Graph permissions enabled: email, offline_access, openid, profile, and User.Read.
```

Comment:
Could you please correct the steps to setup this app? These are steps to configure SSO.

Reply:
Resolved

Reply:
I still see reference of SSO.. 

> Our SSO flow will give you access to the first 4 permissions, and we will have to exchange the token server-side to get an elevated token for the profile permission              (for example, if we want access to the user's profile photo).